File: app/src/main/java/com/absinthe/anywhere_/viewbuilder/entity/CardListDialogBuilder.java
package com.absinthe.anywhere_.viewbuilder.entity;

import android.content.Context;
import android.view.Gravity;
import android.widget.FrameLayout;
import android.widget.TextView;
import android.widget.ViewFlipper;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.absinthe.anywhere_.R;
import com.absinthe.anywhere_.adapter.applist.AppListAdapter;
import com.absinthe.anywhere_.viewbuilder.ViewBuilder;

public class CardListDialogBuilder extends ViewBuilder {

    public AppListAdapter mAdapter;

    public CardListDialogBuilder(Context context) {
        super(context);
    }

    @Override
    public void init() {
        root = new ViewFlipper(mContext);
        root.setLayoutParams(Params.LL.MATCH_MATCH);
        int padding = d2p(10);
        root.setPadding(0, padding, 0, padding);

        RecyclerView rvCardList = new RecyclerView(mContext);

        rvCardList.setLayoutParams(new FrameLayout.LayoutParams(
                FrameLayout.LayoutParams.MATCH_PARENT,
                FrameLayout.LayoutParams.WRAP_CONTENT
        ));
        addView(rvCardList);

        rvCardList.setLayoutManager(new LinearLayoutManager(mContext));
        mAdapter = new AppListAdapter(mContext, AppListAdapter.MODE_CARD_LIST);
        rvCardList.setAdapter(mAdapter);

        TextView tvEmpty = new TextView(mContext);
        tvEmpty.setLayoutParams(Params.LL.MATCH_WRAP);
        tvEmpty.setPadding(padding, padding, padding, padding);
        tvEmpty.setGravity(Gravity.CENTER_HORIZONTAL);
        tvEmpty.setTextSize(24);
        tvEmpty.setTextColor(mResources.getColor(R.color.textColorNormal));
        tvEmpty.setText(R.string.dialog_empty_list);
        addView(tvEmpty);
    }

    public void setOnItemClickListener(AppListAdapter.OnItemClickListener listener) {
        mAdapter.setOnItemClickListener(listener);
    }
}

File: app/src/main/java/com/absinthe/anywhere_/viewbuilder/entity/CardListDialogBuilder.kt
package com.absinthe.anywhere_.viewbuilder.entity

import android.content.Context
import android.view.Gravity
import android.widget.FrameLayout
import android.widget.TextView
import android.widget.ViewFlipper
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.absinthe.anywhere_.R
import com.absinthe.anywhere_.adapter.applist.AppListAdapter
import com.absinthe.anywhere_.viewbuilder.ViewBuilder

class CardListDialogBuilder(context: Context) : ViewBuilder(context) {

    var mAdapter: AppListAdapter = AppListAdapter(mContext, AppListAdapter.MODE_CARD_LIST)

    init {
        root = ViewFlipper(context).apply {
            layoutParams = FrameLayout.LayoutParams(
                    FrameLayout.LayoutParams.MATCH_PARENT,
                    FrameLayout.LayoutParams.MATCH_PARENT
            )
            setPadding(0, 10.dp, 0, 10.dp)
        }

        val rvCardList = RecyclerView(context).apply {
            layoutParams = FrameLayout.LayoutParams(
                    FrameLayout.LayoutParams.MATCH_PARENT,
                    FrameLayout.LayoutParams.WRAP_CONTENT
            )
            layoutManager = LinearLayoutManager(context)
            adapter = mAdapter
        }
        addView(rvCardList)

        val tvEmpty = TextView(context).apply {
            layoutParams = FrameLayout.LayoutParams(
                    FrameLayout.LayoutParams.MATCH_PARENT,
                    FrameLayout.LayoutParams.WRAP_CONTENT
            )
            setPadding(10.dp, 10.dp, 10.dp, 10.dp)
            gravity = Gravity.CENTER_HORIZONTAL
            textSize = 24f
            setTextColor(ContextCompat.getColor(context, R.color.textColorNormal))
            setText(R.string.dialog_empty_list)
        }
        addView(tvEmpty)
    }

    fun setOnItemClickListener(listener: AppListAdapter.OnItemClickListener?) {
        mAdapter.setOnItemClickListener(listener)
    }

    companion object {
        const val VF_LIST = 0
        const val VF_EMPTY = 1
    }
}